wgpu-core/src/command/mod.rs

```diff
@@ -114,6 +114,50 @@ impl CommandEncoderStatus {
         }
     }
 
+    /// Record commands using the supplied closure.
+    ///
+    /// If the encoder is in the [`Self::Recording`] state, calls the closure to
+    /// record commands. If the closure returns an error, stores that error in
+    /// the encoder for later reporting when `finish()` is called. Returns
+    /// `Ok(())` even if the closure returned an error.
+    ///
+    /// If the encoder is not in the [`Self::Recording`] state, the closure will
+    /// not be called and nothing will be recorded. The encoder will be
+    /// invalidated (if it is not already). If the error should be raised
+    /// immediately, returns it in `Err`, otherwise, returns `Ok(())`.
```

Comment:
question: "should be" implies some reference to the spec. Can you provide (even a very general) one here?